components/english/speaking-practice.tsx

```
@@ -0,0 +1,635 @@
+  const [feedback, setFeedback] = useState<"correct" | "incorrect" | "">("");
+  const [pronunciationScore, setPronunciationScore] = useState(0);
+  const [detailedScores, setDetailedScores] = useState<{
+    accuracy: number;
+    fluency: number;
+    completeness: number;
+  }>({ accuracy: 0, fluency: 0, completeness: 0 });
+  const [showDefinition, setShowDefinition] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+  const [showScoreHistory, setShowScoreHistory] = useState(false);
+
+  const recognitionRef = useRef<any>(null);
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)